src/has-yarn.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const path = require('path');
+const fs = require('fs-extra');
+
+/**
+ * Check if there is a yarn.lock file to indicate if the project uses yarn as the package manager
+ *
+ * @param {string} projectRoot - Path to the project root to check for yarn usage
+ * @returns {boolean}
+ */
+module.exports = function hasYarn(projectRoot) {
+  let isYarn = false;
+  try {
+    fs.accessSync(path.join(projectRoot, 'yarn.lock'), fs.constants.F_OK);
```

Comment:
Is there a reason you removed the async it used to have?

Reply:
No particular reason. The function content is sync so I figure it didn't need to be async

Reply:
sync is simpler, and IMHO preferred

Reply:
I prefer async where possible. Then I don't have to think about it.

Reply:
Modified to

```
module.exports = async function hasYarn(projectRoot) {
  let isYarn = false;
  try {
    await fs.access(path.join(projectRoot, 'yarn.lock'), fs.constants.F_OK);
    isYarn = true;
  } catch (err) {}
  return isYarn;
};
```